srml/treasury/src/lib.rs

```diff
@@ -14,7 +14,48 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! The Treasury: Keeps account of the taxed cash and handles its deployment.
+//! # Treasury Module
+//! 
+//! The `treasury` module keeps account of currency in a `pot` and manages the subsequent
+//! deployment of these funds.
+//! 
+//! ## Overview
+//! 
+//! Funds for treasury are raised in two ways:
+//! 1. By minting new tokens, leading to inflation, and
+//! 2. By channelling some of the tokens set for burning from transaction fees and slashing.
+//! 
+//! Treasury funds can be used to pay for developers who provide software updates,
```

Comment:
Opinion: I think it makes sense for this module to have a **Scenarios** section similar to [staking](https://github.com/paritytech/substrate/blob/master/srml/staking/src/lib.rs#L49)

Reply:
In general, I think a module like this would fit well with the structure outlined in staking, although it is a bit simpler.